reactor-core/src/main/java/reactor/core/publisher/UnicastManySinkNoBackpressure.java

```diff
@@ -67,13 +70,38 @@
 	public void subscribe(CoreSubscriber<? super T> actual) {
 		Objects.requireNonNull(actual, "subscribe");
 
-		if (!STATE.compareAndSet(this, State.INITIAL, State.SUBSCRIBED)) {
+		boolean casFails;
+		State s;
+		do {
+			s = state;
+			switch(s) {
+				case TERMINATED_BEFORE_SUBSCRIPTION:
+					casFails = !STATE.compareAndSet(this, s, State.TERMINATED);
+					break;
+				case INITIAL:
+					casFails = !STATE.compareAndSet(this, s, State.SUBSCRIBED);
+					break;
+				default:
+					casFails = false;
+			}
+		} while (casFails);
+
+		if (s == State.TERMINATED_BEFORE_SUBSCRIPTION) {
+			//detected pre-subscribe termination
+			if (this.error != null) {
+				Operators.error(actual, this.error);
+			}
+			else {
+				Operators.complete(actual);
+			}
```

Comment:
why not put it into switch's cases?
Similar to https://github.com/reactor/reactor-core/blob/56e5c0ddf8e9485b9117c75795db0afcd64eef6e/reactor-core/src/main/java/reactor/core/publisher/FluxConcatMapNoPrefetch.java#L276-L286

1. try CAS
2. continue if fails
3. do the action otherwise

Reply:
see 80278ec